agent/app/agent_capability_unix_test.go

```diff
@@ -702,6 +730,104 @@ func TestTaskEIACapabilitiesUnix(t *testing.T) {
 		dockerClient:       client,
 		credentialProvider: aws_credentials.NewCredentials(mockCredentialsProvider),
 		mobyPlugins:        mockMobyPlugins,
+		ioutil:             mockIoutil,
+	}
+	capabilities, err := agent.capabilities()
+	assert.NoError(t, err)
+
+	for i, expected := range expectedCapabilities {
+		assert.Equal(t, aws.StringValue(expected.Name), aws.StringValue(capabilities[i].Name))
+		assert.Equal(t, aws.StringValue(expected.Value), aws.StringValue(capabilities[i].Value))
+	}
+}
+
+func TestTaskEIACapabilitiesWithOptimizedCPU(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_dockerapi.NewMockDockerClient(ctrl)
+	mockMobyPlugins := mock_mobypkgwrapper.NewMockPlugins(ctrl)
+	mockCredentialsProvider := app_mocks.NewMockProvider(ctrl)
+	mockIoutil := mock_ioutilwrapper.NewMockIOUtil(ctrl)
+
+	conf := &config.Config{
+		PrivilegedDisabled: true,
+	}
+
+	fileBytes, _ := ioutil.ReadFile(filepath.Join(".", "testdata", "test_cpu_info"))
+
+	gomock.InOrder(
+		client.EXPECT().SupportedVersions().Return([]dockerclient.DockerVersion{
+			dockerclient.Version_1_17,
+		}),
+		client.EXPECT().KnownVersions().Return([]dockerclient.DockerVersion{
+			dockerclient.Version_1_17,
+		}),
+		mockMobyPlugins.EXPECT().Scan().AnyTimes().Return([]string{}, nil),
+		client.EXPECT().ListPluginsWithFilters(gomock.Any(), gomock.Any(), gomock.Any(),
+			gomock.Any()).AnyTimes().Return([]string{}, nil),
+		mockIoutil.EXPECT().ReadFile(CpuInfoPath).Return(fileBytes, nil),
+	)
+
+	expectedCapabilityNames := []string{
+		"com.amazonaws.ecs.capability.docker-remote-api.1.17",
+	}
+
+	var expectedCapabilities []*ecs.Attribute
+	for _, name := range expectedCapabilityNames {
+		expectedCapabilities = append(expectedCapabilities,
+			&ecs.Attribute{Name: aws.String(name)})
+	}
+	expectedCapabilities = append(expectedCapabilities,
```

Comment:
nit: for this test, can we just do the relevant capabilities? And do an assert.Contains? otherwise we need to update this list in the future.

Reply:
Why doing contains instead of equal?

Reply:
Otherwise you need to modify this list every time we add new capabilities. We can instead only focus on the relevant capabilities